tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -0,0 +1,433 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// This module facilitates testing various connect redirect scenarios by sending traffic to both
+// local system and a remote system, both running TCP / UDP listeners.
+
+#define CATCH_CONFIG_RUNNER
+
+#include "catch_wrapper.hpp"
+
+#include "bpf/bpf.h"
+#pragma warning(push)
+#pragma warning(disable : 4200)
+#include "bpf/libbpf.h"
+#pragma warning(pop)
+#include "common_tests.h"
+#include "ebpf_nethooks.h"
+#include "ebpf_structs.h"
+#include "socket_helper.h"
+#include "socket_tests_common.h"
+
+#include <mstcpip.h>
+
+static std::string _family;
+static std::string _protocol;
+static std::string _vip_v4;
+static std::string _vip_v6;
+static std::string _local_ip_v4;
+static std::string _local_ip_v6;
+static std::string _remote_ip_v4;
+static std::string _remote_ip_v6;
+static uint16_t _destination_port = 4444;
+static uint16_t _proxy_port = 4443;
+
+typedef struct _test_addresses
+{
+    struct sockaddr_storage loopback_address;
+    struct sockaddr_storage remote_address;
+    struct sockaddr_storage local_address;
+    struct sockaddr_storage vip_address;
+} test_addresses_t;
+
+typedef struct _test_globals
+{
+    ADDRESS_FAMILY family;
+    IPPROTO protocol;
+    uint16_t destination_port;
+    uint16_t proxy_port;
```

Comment:
Why are these needed outside the addresses[]?  The ports should be inside them, no?

Reply:
The same addresses are used both as destination and proxy by the test cases, hence storing the proxy and destination port separately.